readthedocs/settings/proxito.py

```diff
@@ -0,0 +1,38 @@
+"""Local setting for Proxito"""
+
+# pylint: disable=missing-docstring
+
+import os
+
+from readthedocs.settings import base
+
+
+class ProxitoDevSettings(base.CommunityBaseSettings):
+
+    ROOT_URLCONF = 'readthedocs.proxito.urls'
+    USE_SUBDOMAIN = True
+
+    PUBLIC_DOMAIN = 'dev.readthedocs.io'
+    PUBLIC_DOMAIN_USES_HTTPS = False
+
+    @property
+    def MIDDLEWARE(self):  # noqa
+        # Use our new middleware instead of the old one
+        classes = super().MIDDLEWARE
+        classes = list(classes)
+        index = classes.index(
+            'readthedocs.core.middleware.SubdomainMiddleware'
+        )
+        classes[index] = 'readthedocs.proxito.middleware.ProxitoMiddleware'
+        classes.remove('readthedocs.core.middleware.SingleVersionMiddleware')
```

Comment:
Something to figure out is the ideal position of this middleware which might vary from .org to .com and which other middlewares should be included with a production proxito deployment.

Ideally, you want it as high up the chain as possible while still being able to handle everything necessary. To handle requests involving authorization, it has to come after SessionMiddleware and AuthenticationMiddleware. However, for RTD Community, where authorization isn't necessary, it should probably be above those (or not include them entirely) to avoid the overhead of auth/sessions in the case where it's just redirecting/proxying to storage/etc.

Reply:
Yea, I don't want to worry too much about optimization at this point. There's definitely a project to reduce the memory usage and processing of proxito in the default case as we move forward to rolling this out more widely.